internal/dtime/deckard_time.go

```diff
@@ -0,0 +1,81 @@
+package dtime
+
+import (
+	"time"
+)
+
+const (
+	millisPerSecond     = int64(time.Second / time.Millisecond)
+	nanosPerMillisecond = int64(time.Millisecond / time.Nanosecond)
+)
+
+var (
+	nowProvider = func() time.Time {
+		return time.Now()
+	}
+	internalTimeMocking []time.Time
+)
+
+func resetNowProvider() {
+	nowProvider = func() time.Time {
+		return time.Now()
+	}
+	internalTimeMocking = nil
+}
+
+// Visible for testing
+// Returns the reset function to restore the default provider
+func SetNowProvider(provider func() time.Time) func() {
+	nowProvider = provider
```

Comment:
if the nowProvider always receives an unmocked function here
![image](https://github.com/takenet/deckard/assets/56096912/60fd0d0f-7756-4f76-8faa-03872976a159)


how is this assignment any useful if the resetNowProvider is return after executing this line ?

Reply:
I don't understood what you mean.

The `resetNowProvider` is a function to be called to reset the `nowProvider` to the original `time.Now()` result.

The `SetNowProvider` will change the now provider and return a function to be called in the end of the test. Usage example:

```
		defer SetNowProvider(func() time.Time {
			return time.Now().Add(time.Second)
		})()
```

This will mock the `nowProvider` to return always `now + 1` second and defer the reset of the provider after the end of the current block.

Reply:
I meant what was the point of doing this assignment in this line if the returned value always is the `resetNow` function and nothing related to this line

Reply:
But I see now that is the case because you defer the reset after a mock, nicely done.